easistrain/EDD/fitEDD.py

```diff
@@ -352,15 +358,21 @@ def fitEDD(
                     maxBoundsVD = np.append(
                         maxBoundsVD, appendMaxBoundsVD
                     )  # maximum bounds of the parametrs solution (H, C, FWHM1, FWHM2, eta) for the vertical detector
-                optimal_parametersVD, covarianceVD = scipy.optimize.curve_fit(
-                    f=splitPseudoVoigt,
-                    xdata=peakVerticalDetector[:, 0],
-                    ydata=peakVerticalDetector[:, 1] - yCalculatedBackgroundVD,
-                    p0=initialGuessVD,
-                    sigma=None,
-                    bounds=(minBoundsVD, maxBoundsVD),
-                    maxfev=10000,
-                )  ## fit of the peak of the Vertical detector
+                try:
+                    optimal_parametersVD, covarianceVD = scipy.optimize.curve_fit(
+                        f=splitPseudoVoigt,
+                        xdata=peakVerticalDetector[:, 0],
+                        ydata=peakVerticalDetector[:, 1] - yCalculatedBackgroundVD,
+                        p0=initialGuessVD,
+                        sigma=None,
+                        bounds=(minBoundsVD, maxBoundsVD),
+                        maxfev=10000,
+                    )  ## fit of the peak of the Vertical detector
+                except RuntimeError:
+                    optimal_parametersVD = np.ones_like(initialGuessVD)
+                    covarianceVD = np.ones(
+                        (5 * nbPeaksInBoxes[i], 5 * nbPeaksInBoxes[i])
+                    )
```

Comment:
Q: Is the problem that if a fit fails, it completely stops the program ?
A: Yes.

Q: Perhaps we should instead completely skip the saving of the fit ?
A: It can be done like that also but then we have to adapt all the rest of the function to that modification. The idea by putting a matrix of ones is that if the fit fails the user of the routine will see immediately that there is something wrong with that point when plotting the results for example. 

In any case, I think it is worth putting a print to warn the user --> Totally agree with you for this point. We must do it.

Reply:
Zeros, I would avoid because it can cause some other issues for the next tasks as for example dividing by zero ....
NaN, it can also be an option but I do not know if it could also crush the next steps if we want to do operations on NaN values.

Reply:
Normally, operations on `NaN` will yield `NaN`. Which is good because we will be able to know in subsequent steps that a fit failed.

Then, at some point in the processing (when regrouping?), we should filter out `NaN` to avoid processing the data where the fit failed.